src/meta/src/controller/catalog.rs

```diff
@@ -0,0 +1,385 @@
+            .chain(indexes.into_iter().map(|i| i.index_id))
+            .collect::<Vec<_>>();
+        let source_ids = db
+            .find_related(Source)
+            .all(&txn)
+            .await?
+            .into_iter()
+            .map(|s| s.source_id)
+            .collect_vec();
+        let connections = db
+            .find_related(Connection)
+            .all(&txn)
+            .await?
+            .into_iter()
+            .flat_map(|c| c.info)
+            .collect_vec();
+
+        let mut query_schema = Query::select()
+            .column(schema::Column::SchemaId)
+            .from(Schema)
+            .and_where(schema::Column::DatabaseId.eq(database_id as i32))
+            .to_owned();
+        let query_table = Query::select()
+            .column(table::Column::TableId)
+            .from(Table)
+            .and_where(table::Column::DatabaseId.eq(database_id as i32))
+            .to_owned();
+        let query_source = Query::select()
+            .column(source::Column::SourceId)
+            .from(Source)
+            .and_where(source::Column::DatabaseId.eq(database_id as i32))
+            .to_owned();
+        let query_sink = Query::select()
+            .column(sink::Column::SinkId)
+            .from(Sink)
+            .and_where(sink::Column::DatabaseId.eq(database_id as i32))
+            .to_owned();
+        let query_index = Query::select()
+            .column(index::Column::IndexId)
+            .from(Index)
+            .and_where(index::Column::DatabaseId.eq(database_id as i32))
+            .to_owned();
+        let query_view = Query::select()
+            .column(view::Column::ViewId)
+            .from(View)
+            .and_where(view::Column::DatabaseId.eq(database_id as i32))
+            .to_owned();
+        let query_function = Query::select()
+            .column(function::Column::FunctionId)
+            .from(Function)
+            .and_where(function::Column::DatabaseId.eq(database_id as i32))
+            .to_owned();
+        let query_connection = Query::select()
+            .column(connection::Column::ConnectionId)
+            .from(Connection)
+            .and_where(connection::Column::DatabaseId.eq(database_id as i32))
+            .to_owned();
+        let query_all = query_schema
+            .union(UnionType::All, query_table)
+            .union(UnionType::All, query_source)
+            .union(UnionType::All, query_sink)
+            .union(UnionType::All, query_index)
+            .union(UnionType::All, query_view)
+            .union(UnionType::All, query_function)
+            .union(UnionType::All, query_connection)
+            .union(
+                UnionType::All,
+                Query::select()
+                    .expr(Expr::value(database_id as i32))
+                    .to_owned(),
+            )
+            .to_owned();
+        // drop related objects, the relations will be dropped cascade.
+        Object::delete_many()
```

Comment:
Is it possible to make `database_id` a foreign key and delete them in a cascade way?

Reply:
Currently I used object id to delete them in a cascade way, the `database_id` is actually a foreign key referring to object table. 🤔 Didn't find a easier way to do in a cascade way by `database_id` yet.